integration-docs: Update to the new format (PagerDuty, Papertrail, Patreon, Pingdom).

adnan-td opened the pull request:
Apply integration-docs' new format.

Fixes part of #29592.

Updates the following integration documentation pages:

<details>
<summary>PagerDuty</summary>

[Doc link](https://zulip.com/integrations/doc/pagerduty)

![localhost_9991_integrations_doc_pagerduty](https://github.com/user-attachments/assets/cd0ff913-75fb-4899-9333-69ab9c6dc8e2)

</details>

<details>
<summary>Papertrail</summary>

[Doc link](https://zulip.com/integrations/doc/papertrail)

![localhost_9991_integrations_doc_papertrail](https://github.com/user-attachments/assets/b8225350-67b4-43e0-b6a1-d0644376031c)

</details>

<details>
<summary>Patreon</summary>

[Doc link](https://zulip.com/integrations/doc/patreon)

![localhost_9991_integrations_doc_patreon (1)](https://github.com/user-attachments/assets/b9180012-68aa-439d-bb27-d9ee1ad3314e)

</details>

<details>
<summary>Pingdom</summary>

[Doc link](https://zulip.com/integrations/doc/pingdom)

![localhost_9991_integrations_doc_pingdom (2)](https://github.com/user-attachments/assets/e76db15d-4e3b-41de-ada0-9c0faf7ffbc6)

</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

adnan-td commented on zerver/webhooks/pingdom/doc.md:
```diff
@@ -1,24 +1,33 @@
+    menu, and click on **Add integration**.
 
-    `https://my.pingdom.com/reports/integration/settings`
-
```
Replaced the link which was confusing with text for clearity.

adnan-td commented on zerver/webhooks/pingdom/doc.md:
```diff
@@ -1,24 +1,33 @@
 
     ![](/static/images/integrations/pingdom/001.png)
 
-1. Last, during creating or editing your check, scroll down to
-    **Connect Integrations** section and ensure your integration is checked.
+1. Ensure your integration is selected in the **Check** settings.
```
Referred the docs here: [doc1](https://support.atlassian.com/opsgenie/docs/integrate-opsgenie-with-pingdom/), [doc2](https://www.pagerduty.com/docs/guides/pingdom-integration-guide/)

adnan-td commented:
@PieterCK and @Niloth-p this is ready to be checked out!

PieterCK commented:
No comment from me! changes looks good :+1:

Niloth-p commented on zerver/webhooks/pagerduty/doc.md:
```diff
@@ -1,20 +1,32 @@
-1. Set **Webhook URL** to the URL constructed above,
+1. Set **Webhook URL** to the URL generated above,
    configure the **Scope Type** and **Scope** you want to receive notifications for,
    select the events you want to send under **Events to Send**,
```
We've been adding a link to the "Filtering events" section in docs with similar instructions, for the word "events".

Niloth-p commented on zerver/webhooks/patreon/doc.md:
```diff
@@ -1,22 +1,35 @@
 
 1. Login to your Patreon developer account, navigate to your Patreon Portal, and
-   click on [My Webhooks](https://www.patreon.com/portal/registration/register-webhooks).
+    click on [My Webhooks](https://www.patreon.com/portal/registration/register-webhooks).
```
Use a reference link instead?

adnan-td replied:
Not sure what's a reference link. Do you mean the related documentation section?

Niloth-p replied:
I meant this.
```
[My Webhooks][1]

[1]: URL
```
Put the ref links at the bottom of the document.

Niloth-p commented on zerver/webhooks/patreon/doc.md:
```diff
@@ -1,22 +1,35 @@
+    click on [My Webhooks](https://www.patreon.com/portal/registration/register-webhooks).
+
+1. Paste the URL generated above in the webhook URL field, and click the
+    **plus** (**+**) button. Enable the events you would like to receive
```
Similar to the other comment, add link on "events".

Niloth-p commented on zerver/webhooks/pingdom/doc.md:
```diff
@@ -1,24 +1,33 @@
+1. {!generate-webhook-url-basic.md!}
 
-1. Next, go to the following URL:
+1. In the main menu, go to **Integrations** under the **Integrations**
```
Good improvement indeed.

```suggestion
1. From the main menu, open the "Integrations" menu, and
```
To make the order of the instructions same as the traversal steps, as well as conformity.

Niloth-p commented on zerver/webhooks/pingdom/doc.md:
```diff
@@ -1,24 +1,33 @@
-
-1. Create your integration by clicking on the **Add Integration** button
-    and filling the form like so:
+1. Configure the integration as follows and click **Save integration**:
```
I think it might be better to verbalize these settings, and remove the image.
For example: people who use screen readers might find it helpful this way.

Niloth-p reviewed (commented):
Lgtm, Adnan!
Just added a few suggestions to conform better with other PRs.

I didn't look into the correctness of the Pingdom instructions, but I agree that it was great to replace the link with an instruction, and I think you could do the same for the image as well.

adnan-td commented:
Updated Pingdom docs as suggested by @Niloth-p. Replaced the image to show **Add Integration** with text
instructions.

<details>
<summary>Pingdom</summary>

![image](https://github.com/user-attachments/assets/6aa7923f-7c1e-448f-a9dd-d897acd94721)
</details>

Niloth-p reviewed (commented):
LGTM!

adnan-td commented:
Thanks @Niloth-p! Addressed all reviews.

@laurynmm this PR is ready for further review.

laurynmm commented:
@adnan-td - Thanks for working on updating these integration documentation pages! I made a few revisions to your changes and pushed those here. See updated screenshots below.

<details>
<summary>Pingdom</summary>

[current documentation](https://zulip.com/integrations/doc/pingdom)
![Screenshot from 2024-08-12 18-05-05](https://github.com/user-attachments/assets/455e09e6-cc55-429e-8cad-3d7f5d514888)
</details>
<details>
<summary>Patreon</summary>

[current documentation](https://zulip.com/integrations/doc/patreon)
![Screenshot from 2024-08-12 18-05-23](https://github.com/user-attachments/assets/13a91ce1-170d-4a71-aa79-895b41acf5bd)
</details>
<details>
<summary>PagerDuty</summary>

[current documentation](https://zulip.com/integrations/doc/pagerduty)
![Screenshot from 2024-08-12 18-05-34](https://github.com/user-attachments/assets/5d746e0e-1d98-46c0-9c5a-d46585b13923)
</details>
<details>
<summary>Papertrail</summary>

[current documentation](https://zulip.com/integrations/doc/papertrail)
![Screenshot from 2024-08-12 18-05-44](https://github.com/user-attachments/assets/c7da12dd-a655-4bb5-a38a-23b3ad81b05e)
</details>

timabbott commented:
Merged, thanks @adnan-td and review team!

timabbott commented:
This was backported in #31355.
